src/tasks/WasmAppBuilder/SignatureMapper.cs

```diff
@@ -51,19 +58,34 @@ internal static class SignatureMapper
                 c = 'I';
             else if (t.IsInterface)
                 c = 'I';
-            else if (t.IsEnum)
-                c = TypeToChar(t.GetEnumUnderlyingType(), log);
-            else if (t.IsPointer)
+            else if (t.IsEnum) {
+                Type underlyingType = t.GetEnumUnderlyingType();
+                if (underlyingType != t)
+                    c = TypeToChar(underlyingType, log, out _, ++depth);
+                else {
+                    log.Warning("WASM0064", $"Unsupported parameter type '{t.Name}'");
```

Comment:
When can this happen?

Reply:
I don't know of any realistic scenario where unbounded recursion can occur in TypeToChar, but it broke the wasm perf measurements so I am doing everything I can to fix it. I can't repro it locally.